File: src/main/java/it/contrader/dto/ReviewDTO.java
package it.contrader.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import it.contrader.model.Esercizio;
import it.contrader.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class ReviewDTO {
	
	private Long id;
	
	
	private String titolo;
	
	
	private String testo;
	
	
	private Integer voto;
	
	
	private User user;
	
	private Esercizio esercizio;


}

File: src/dto/reviewdto.ts
import { UserDTO } from './userdto';
import { EsercizioDTO } from './eserciziodto';



export class ReviewDTO{

 id: number;


titolo: string;


testo: string;


voto: number;

esercizioId: number;


 esercizio: EsercizioDTO;

 esercizioNome: string;
}
